### PR TITLE
fix(react-grid): calculate actual row count in VirtualTableState

### LINKED ISSUE #2876: React Grid doesn't render all rows after filtering if VirtualTable is used
- [x] I have searched this repository's [issues](https://github.com/devexpress/devextreme-reactive/issues) and believe that this is not a duplicate.

### I'm using ...

- [x] React Grid
- [ ] React Chart
- [ ] React Scheduler

###  Example: 
https://codesandbox.io/s/musing-napier-ywfs7?file=/demo.js

###  Steps:
1. Write E200 to the Product column and click Enter
2.  Remove E200 and click Enter

### Current Behaviour

React Grid doesn't correctly render the virtual table and displays gray rectangles

### Expected Behaviour

React Grid should render all rows

### LINKED ISSUE #2877: VirtualTable doesn't render last 3 cells, eg if totalRowCount === 103
- [x ] React Grid

consider the row caching example from the docs - https://devexpress.github.io/devextreme-reactive/react/grid/docs/guides/lazy-loading/#row-caching

I changed line number 195 by enforcing total rows to **103** rows. 
https://codesandbox.io/s/awesome-drake-w5zqh?file=/demo.js
scroll to the end of the table
The table won't trigger additional request to bring the last 3 cells.
The table show the skeletons cells instead

Please investigate ASAP, we are a paying customer!

### Current Behaviour
Only 100 rows fetched / shown in table

### Expected Behaviour
Should bring and show all 103 rows with virtual scrolling
It may also fail for other totals, 110, 120 etc... 

### Steps to Reproduce
check the description and example above

### Environment
Mac + Chrome

- devextreme-reactive: view your example
- react: xxx
- browser: chrome macOS
- material-ui: xxx

### PR DESCRIPTION
**Pull request by LazyLahtak**
Fixes #2876
Fixes #2877

### CODE REVIEW

**Comment by LazyLahtak** [comment]
Ready to review

File: packages/dx-react-grid/src/plugins/virtual-table/virtual-table-state.tsx [hunk 1]
```diff
@@ -116,10 +118,18 @@ class VirtualTableStateBase extends React.PureComponent<VirtualTableStateProps,
...
     } = nextProps;
 
     return {
-      availableRowCount,
+      availableRowCount: getAvailableRowCount(
+        infiniteScrolling,
+        pageSize * 2,
+        availableRowCount,
+        totalRowCount,
+      ),
```
Comment (MaximKudriavtsev) [inline packages/dx-react-grid/src/plugins/virtual-table/virtual-table-state.tsx:132]: At line 156 `const availableRowCount = infiniteScrolling ? stateRowCount : totalRowCount;`

Is this necessary code? Seems that the `getAvailableRowCount` function has this statement

**Comment by artyil** [comment]
Hi, will there be  a patch release with this fix?

**Comment by MaximKudriavtsev** [comment]
Hi @artyil,

The fix will be published soon.

**Pull request merged**